packages/platform-sdk-profiles/src/environment/services/exchange-rate-service.ts

```diff
@@ -29,6 +29,10 @@ export class ExchangeRateService {
 				.filter((wallet: ReadWriteWallet) => wallet.currency() === currency);
 		}
 
+		if (!wallets.length || !wallets[0].network().isLive()) {
```

Comment:
Can't we already filter out `isTest` wallets before we even filter the currency? Then the second filter would be a noop if there are no live wallets for that currency.

Reply:
Right now the methods for the exchange rate are based on `allByCoin`, whose purpose is questionable at this point, as it groups live & test wallets together.